### DISCUSSION TITLE
Use with non-ASUS laptops

### DISCUSSION THREAD

**Discussion by logicito** [tshakalekholoane/bat#14] [Q&A] [answered]
I have an MSI Summit E13 Flip (Intel EVO certified) with Ubuntu 21.04, will this utility work?

Because I use the laptop mostly plugged to the AC adapter, and to prolong the battery life, I have setup in Windows the battery charging thresholds (within MSI Center for Business and Productivity) to:

- Charge the battery only when is below 50%, then charge it up to 60%

I use the laptop (MSI Summit E13 Flip EVO) with Windows and Ubuntu, mostly Ubuntu, I haven't been able to find a way to manage the battery thresholds from Ubuntu. I was exploring the advanced BIOS settings, activaded by Right Shift + Right Ctrl + Left Alt + F2, looking for an option to manage it, so far nothing.

Any idea how this can be configured from outside Windows? Long term I would like to completely remove the Windows partition and use only Ubuntu, right now I can't do it (maybe with bat), because I need Windows and the MSI Center for Business and Productivity to configure the Battery Thresholds.

NOTE: The battery thresholds settings are saved at BIOS level (EC table), because once set in Windows they will remain in Ubuntu and even while inside the BIOS.

**Comment by tshakalekholoane** [comment, marked as answer]
In its current state, I don't think this utility will be able to set the charging threshold on your laptop. Reason being, it relies on abstractions provided by the Linux operating system, and by the looks of things [1][2], this does not seem to be supported for MSI systems.

I don't know how else this could be achieved outside of the utility provided by the manufacturer in Windows.

- [1] https://github.com/torvalds/linux/blob/master/drivers/platform/x86/msi-wmi.c
- [2] https://github.com/torvalds/linux/blob/master/drivers/platform/x86/msi-laptop.c
